File: AspNetCoreSpa.WebApi/ClientApp/src/app/models/post.ts
import { User } from './user';
import { Image } from './image';
import { Like } from './like';

export class Post {
    id: number;
    title: string;
    description: string;
    createAt: string;
    updateAt: string;
    images: Image[];
    user: User;
    countLike: number;
    likes: Like[];
}

File: AspNetCoreSpa.Domain/Entities/Post.cs
using System;
using System.Collections.Generic;
using AspNetCoreSpa.Domain.Entities.Base;

namespace AspNetCoreSpa.Domain.Entities
{
    public class Post : BaseEntity<int>
    {
        public Post()
        {
            Images = new HashSet<PostImage>();
            Comments = new HashSet<Comment>();
            Likes = new HashSet<Like>();
        }

        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreateAt { get; set; }
        public DateTime? UpdateAt { get; set; }
        public User User { get; set; }
        public ICollection<PostImage> Images { get; set; }
        public ICollection<Comment> Comments { get; set; }
        public ICollection<Like> Likes { get; set; }
    }
}